### PR TITLE
Bug 1156089 - Fix remaining fl jshint issues

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1156089

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=af4fe131d6c4e0549e3e73577d8a910788ccc10e

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=29cf6bc304a3cb622ef3b6a04b93156cafac30df

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b23e525cbd356623acb5c0f4357dcdade4bf5a4d

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e2e694b6886ac45b606158904b5ef497d19bb053

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=331f727c486beb79520a9056a0820961259df595

File: apps/fl/js/utils.js [hunk 1]
```diff
@@ -15,6 +25,16 @@ function _(id, args) {
 
 // Show a dialog with one or two buttons
 function showDialog(options) {
+
+  function close(e) {
+    document.body.removeChild(dialog);
+    e.preventDefault();
+    e.stopPropagation();
+  }
```
Comment (davidflanagan) [inline apps/fl/js/utils.js]: Does jshint really complain about forward references to functions? I prefer to put utility functions like this at the end where they don't clutter things up and obscure the main purpose of the code. It is not worth changing your patch over, but I'm guessing that you were imposing your own style views here, not just making jshint happy...

Comment (KevinGrandon) [inline apps/fl/js/utils.js]: Nope, it really complains =(

The other option is to declare a var at the top, and assign it later on. E.g.,

```
var close;
function myFunc() {
  // Code that references close.
}
close = function() {};
myFunc();
```

I don't think that is much better, so I think we should go with this for now. Assigning functions to some containing object is also another workaround, but this is the smaller change,

**Pull request merged**